sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -1375,6 +1375,16 @@ object SQLConf {
     .booleanConf
     .createWithDefault(false)
 
+  val FALL_BACK_TO_HDFS_FOR_STATS_MAX_PART_NUM =
+    buildConf("spark.sql.statistics.fallBackToHdfs.maxPartitionNum")
+    .doc("If the number of table partitions exceed this value, falling back to hdfs " +
+      "for statistics calculation is not allowed. This is used to avoid calculating " +
+      "the size of a large number of partitions through hdfs, which is very time consuming." +
+      "Setting this value to 0 or negative will disable falling back to hdfs for " +
+      "partition statistic calculation.")
```

Comment:
does the data source table have the same problem?

Reply:
Yes, in PruneFileSourcePartitions, it also may lead to calculating size of large number of partitions through hdfs.
I will create a follow-up PR to refine it.

Reply:
If this is a common problem, let's leave it here and open a new PR to fix it completely later.

Reply:
yea, will create new PR for it. Already removed from this PR. thanks.